### PR TITLE
Improve QR page

### LINKED ISSUE #66: QR pages design improvements
- Spacing between components
- Heading hierarchy
- Stamp type removal
- TODO text removal
- Presence signature feedback (success info box)
- Presence signature requirements (blue info box)

### PR DESCRIPTION
**Pull request by poiuztrewq1**
A few improvements happened in this pr:
- Added info box on token collection
- removed todo text
- removed types from stamp
- improved spacing
- used common components like Loading, and ServiceContext throw error
- fixed a little bug in the common loading component
- improved the stamp colors, so they are more inline with the style of the page and better to look at in night mode 

closes #66

### CODE REVIEW

**Comment by poiuztrewq1** [comment]
Some questions: Should we handle token types on frontend? Like other icon on tokens with 'extra type' and/or exclude them from counting?

**Comment by Gerviba** [comment]
> Some questions: Should we handle token types on frontend? Like other icon on tokens with 'extra type' and/or exclude them from counting?

We count the 'default' types as stamps, so for testing and other usage we might introduce another icon or a color, but it is not a feature request in any means.

**Comment by berenteb** [comment]
`Error {name: "NoVideoInputDevicesError", message: "No video input devices found"}`

File: src/main/client/src/components/@pages/QRList.tsx [hunk 1]
```diff
@@ -39,32 +41,52 @@ export const QRList: React.FC = (props) => {
...
 
   return (
     <Page {...props} loginRequired>
-      <Heading>QR kód vadászat</Heading>
+      <Heading as="h1">QR kód vadászat</Heading>
+      {alertBar(tokens.length, minTokenToComplete)}
       <Paragraph>
-        A standoknál végzett aktív tevékenyégért QR kódokat lehet gyűjteni, amiket tanköri hiányzások igazolására lehet beváltani. TODO:
-        Fogalmazzunk meg ide szöveget, mert ma bevettem a fogalmazás gátlóm.
+        A standoknál végzett aktív tevékenyégért QR kódokat lehet gyűjteni. Ha eleget összegíűjtesz, beválthatod egy tanköri jelenlétre.
```
Comment (berenteb) [inline src/main/client/src/components/@pages/QRList.tsx:69]: Typoooo

File: src/main/client/src/components/@pages/QRList.tsx [hunk 2]
```diff
@@ -39,32 +41,52 @@ export const QRList: React.FC = (props) => {
...
+      <ButtonGroup mt="5">
         <LinkButton colorScheme="brand" leftIcon={<FaQrcode />} href="/qr/scan">
           QR kód beolvasása
         </LinkButton>
       </ButtonGroup>
 
-      <Heading>Haladás</Heading>
-      <Heading as="h4" size="md">
-        Eddig beolvasott tokenek: {tokens.length} / {totalTokenCount}
+      <Heading as="h3" mt="10">
```
Comment (berenteb) [inline src/main/client/src/components/@pages/QRList.tsx:78]: Add size explicitly. There is no global size definitions for headings for some reasons. Add a lg or smth. Now it maatches size with h1.

File: src/main/client/src/components/@pages/QRScan.tsx [hunk 3]
```diff
@@ -67,47 +57,33 @@ export const QRScan: React.FC = (props) => {
...
         <Fade in={state.state == ScanViewState.Success}>
           <QRScanResultComponent response={state.response || { status: ScanStatus.WRONG }} />
         </Fade>
       )}
-      {state.state !== ScanViewState.Loading && (
-        <ButtonGroup alignSelf="center">
-          <Button marginTop="3" leftIcon={<FaArrowLeft />} onClick={backButtonHandler} size="lg">
-            Vissza
+
+      <ButtonGroup alignSelf="center">
```
Comment (berenteb) [inline src/main/client/src/components/@pages/QRScan.tsx:77]: Easier to add margin top to the button group, not to the buttons individually.

**Review by berenteb — CHANGES_REQUESTED** [review]
It is coming together, appreciate the spacing! Only a few things I noticed. I can't test the QR scan result, just imagined it from the code 😆

**Comment by poiuztrewq1** [comment]
Apart from the requests there are some updates. 
* Cleaned up the states in the QRList page. (Added interface instead of many useState calls)
* Add different icon to the stamp, if its not a "default". So there will be no confusion why there isn't any progress
* Only count default stamps in the progress bar and info box

There are some pictures from the scan page for those who can't run the backend:
Screen Shot 2022-01-26 at 18 25 21
Screen Shot 2022-01-26 at 18 24 29

**Review by berenteb — APPROVED** [review]
Thank you! Also these pictures helped a lot, looks amazing!

**Pull request merged**